homeassistant/util/async_.py

```diff
@@ -93,13 +95,25 @@ def run_callback() -> None:
 
 
 def check_loop(
-    func: Callable[..., Any], strict: bool = True, advise_msg: str | None = None
+    func: Callable[..., Any],
+    strict: bool = True,
+    strict_core: bool = False,
+    advise_msg: str | None = None,
+    **mapped_args: Any,
 ) -> None:
     """Warn if called inside the event loop. Raise if `strict` is True.
 
     The default advisory message is 'Use `await hass.async_add_executor_job()'
     Set `advise_msg` to an alternate message if the solution differs.
     """
+    if (
```

Comment:
This feels very hacky to add logic for a single use case in here. Can we somehow extract this logic out. Maybe allow passing in an lambda to test if a function call should not be checked?

Reply:
Yeah I don't really like it and we have similar logic for sleep on https://github.com/home-assistant/core/pull/114488/files#diff-49c8336cd9b33b91429a785fefd0c00bdd2eadd500adda4c13c91fc124698e90R139

I'm planning on coming back to it once I finish testing

Reply:
I need to fix that late import as well here because it makes everything more complex

Reply:
0adbf28bfbb5c7bfc2ef4f859767de2729ed1bdb